source/PlayerInfo.cpp

```diff
@@ -1821,20 +1827,28 @@ bool PlayerInfo::TakeOff(UI *ui, const bool distributeCargo)
 				cargo.Transfer(outfit.first, outfit.second, Storage());
 			}
 	}
+
 	accounts.AddCredits(income);
 	cargo.Clear();
 	stockDepreciation = Depreciation();
-	if(sold)
+	if(leftOver && income)
```

Comment:
I'd refactor this `if` a bit (see my next comments)
```suggestion
	if(leftOver)
```

Reply:
Yes, that is indeed better.